windows/src/engine/keyman32/DebugEventTrace.cpp

```diff
@@ -0,0 +1,128 @@
+
+#include "pch.h"
+#include "keyman-debug-etw.h"
+
+#ifdef _WIN64
+#define DEBUG_PLATFORM_STRING "x64"
+#define DEBUG_PLATFORM_STRINGW L"x64"
+#else
+#define DEBUG_PLATFORM_STRING "x86"
+#define DEBUG_PLATFORM_STRINGW L"x86"
+#endif
+
+#define TAB L"\t"
+
+
+int GetActualShiftState() {   // I4843
+  int state = 0;
+  if (GetKeyState(VK_NUMLOCK) & 1) state |= NUMLOCKFLAG;
+  if (GetKeyState(VK_CAPITAL) & 1) state |= CAPITALFLAG;
+  if (GetKeyState(VK_SHIFT) < 0) state |= K_SHIFTFLAG;
+  if (GetKeyState(VK_RCONTROL) < 0) state |= RCTRLFLAG;
+  if (GetKeyState(VK_LCONTROL) < 0) state |= LCTRLFLAG;
+  if (GetKeyState(VK_LMENU) < 0) state |= LALTFLAG;
+  if (GetKeyState(VK_RMENU) < 0) state |= RALTFLAG;
+
+  if (GetKeyState(VK_LSHIFT) < 0) state |= 0x10000;
+  if (GetKeyState(VK_RSHIFT) < 0) state |= 0x20000;
```

Comment:
Can these be defined as `LSHIFTFLAG`  and `RSHIFTFLAG`?

Reply:
This is the only place they are used -- for debugging purposes. I don't want to define them because they can't be used :)